rekcurd/core/rekcurd_worker.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+
+from abc import ABCMeta, abstractmethod
+from enum import Enum
+from typing import List, Tuple, Generator
+
+from rekcurd.utils import RekcurdConfig, PredictInput, PredictResult, EvaluateResult, EvaluateDetail, EvaluateResultDetail
+from rekcurd.logger import SystemLoggerInterface, ServiceLoggerInterface, JsonSystemLogger, JsonServiceLogger
+from rekcurd.data_servers import DataServer
+
+
+class Rekcurd(metaclass=ABCMeta):
+    """
+    Rekcurd
+    """
+    _system_logger: SystemLoggerInterface = None
+    _service_logger: ServiceLoggerInterface = None
+    config: RekcurdConfig = None
+    data_server: DataServer = None
+    predictor: object = None
+
+    @abstractmethod
+    def load_model(self, filepath: str) -> object:
+        """
+        load_model
+        :param filepath: ML model file path. str
+        :return predictor: Your ML predictor object. object
+        """
+        raise NotImplemented()
+
+    @abstractmethod
+    def predict(self, predictor: object, idata: PredictInput, option: dict = None) -> PredictResult:
+        """
+        predict
+        :param predictor: Your ML predictor object. object
+        :param idata: Input data. PredictInput, one of string/bytes/arr[int]/arr[float]/arr[string]
+        :param option: Miscellaneous. dict
+        :return result: Result. PredictResult
+            result.label: Label. One of string/bytes/arr[int]/arr[float]/arr[string]
+            result.score: Score. One of float/arr[float]
+            result.option: Miscellaneous. dict
+        """
+        raise NotImplemented()
+
+    @abstractmethod
+    def evaluate(self, predictor: object, filepath: str) -> Tuple[EvaluateResult, List[EvaluateResultDetail]]:
+        """
+        evaluate
+        :param predictor: Your ML predictor object. object
+        :param filepath: Evaluation data file path. str
+        :return result: Result. EvaluateResult
+            result.num: Number of data. int
+            result.accuracy: Accuracy. float
+            result.precision: Precision. arr[float]
+            result.recall: Recall. arr[float]
+            result.fvalue: F1 value. arr[float]
+            result.option: Optional metrics. dict[str, float]
+        :return detail[]: Detail result of each prediction. List[EvaluateResultDetail]
+            detail[].result: Prediction result. PredictResult
+            detail[].is_correct: Prediction result is correct or not. bool
+        """
+        raise NotImplemented()
```

Comment:
Why `predict` and `evaluate` method have `predictor` in parameters, not to just use `self.predictor` ?

Reply:
Since users may not know which variables they can use when they implement `Rekcurd` interface class, adding arguments is understandable for users I believe.

Reply:
I have removed a class variable `predictor` from `Rekcurd` interface class as you mentioned by person. https://github.com/rekcurd/rekcurd-python/pull/36/commits/d5cb9aff24d3fa5a618c68a529774f4d59ddbadb